iceoryx_examples/iceperf/README.md

```diff
@@ -47,248 +51,116 @@ The counter can differ depending on startup of the applications and the performa
-by setting the second parameter of `allocateChunk(...)` to `true`. If not set to `true`,
-a changed payload, while being subscribed, is assumed to be an error.
-
-```cpp
-    for (const auto payloadSizeInKB : payloadSizesInKB)
-    {
-        auto payloadSizeInBytes = payloadSizeInKB * 1024;
-        auto sample = static_cast<PerfTopic*>(myPublisher.allocateChunk(payloadSizeInBytes, true));
-
-        // Specify the payload size for the measurement
-        sample->payloadSize = payloadSizeInBytes;
-        sample->run = true;
-
-        // Send the initial sample to start the round-trips
-        myPublisher.sendChunk(sample);
-
-        // perform the actual measurement
-        auto latency = measureLatency(myPublisher, mySubscriber);
-        latencyInMicroSeconds.push_back(latency);
-
-        // Wait for hardy to send the last response
-        const void* receivedChunk;
-        while (!mySubscriber.getChunk(&receivedChunk))
-        {
-            // poll as fast as possible
-        }
-        mySubscriber.releaseChunk(receivedChunk);
-    }
-```
-
-The actual measurement is performed in the `measureLatency(...)` function. For each round-trip,
-at first the subscriber is polling for the chunk, then it allocates a new chunk with the payload
-specified in the received `PerfTopic` chunk, responds with that chunk and releases the old chunk.
-The time for all round-trips is measured and the average latency is returned.
-
-```cpp
-    double measureLatency(iox::popo::Publisher& publisher, iox::popo::Subscriber& subscriber)
-    {
-        auto start = std::chrono::high_resolution_clock::now();
-        // run the performance test
-        for (uint64_t i = 0; i < NUMBER_OF_ROUNDTRIPS; ++i)
-        {
-            const void* receivedChunk;
-            while (!subscriber.getChunk(&receivedChunk))
-            {
-                // poll as fast as possible
-            }
-
-            auto receivedSample = static_cast<const PerfTopic*>(receivedChunk);
-
-            auto sendSample = static_cast<PerfTopic*>(publisher.allocateChunk(receivedSample->payloadSize, true));
-            sendSample->payloadSize = receivedSample->payloadSize;
-            sendSample->run = true;
-
-            publisher.sendChunk(sendSample);
-
-            subscriber.releaseChunk(receivedChunk);
-        }
-
-        auto finish = std::chrono::high_resolution_clock::now();
-
-        constexpr uint64_t TRANSMISSIONS_PER_ROUNDTRIP{2};
-        auto duration = std::chrono::duration_cast<std::chrono::nanoseconds>(finish - start);
-        auto latencyInNanoSeconds = (duration.count() / (NUMBER_OF_ROUNDTRIPS * TRANSMISSIONS_PER_ROUNDTRIP));
-        auto latencyInMicroSeconds = latencyInNanoSeconds / 1000.;
-        return latencyInMicroSeconds;
-    }
-```
-
-After the benchmark, `Hardy` needs to be shut down. This is done by setting the `run` flag to `false` on the last message to `Hardy`.
+    ****** UNIX DOMAIN SOCKET ********
```

Comment:
I liked the explanation how the performance test was done. Was there a reason to remove it?

Reply:
There were several reasons. It changed completely and now contains uds and mq. Explaining all these is a bit too much IMHO. We could concentrate on the iceoryx part. But there are better examples when it comes to explaining the iceoryx API. And there will be an API breaking change in the next weeks. So we would have to adapt it soon again if we would describe the iceoryx part of iceperf. The last thing I had in mind would be worth to explain is the technology independent part of iceperf (as it is now in laurel and hardy). Do you think we should go for it @dkroenke @elfenpiff @mossmaurice ? It would describe more how iceperf works and not how iceoryx can be used

Reply:
Yes that sounds fine for me

Reply:
I would say go for it @budrus